lib/modules/accountSwitcher.js

```diff
@@ -434,7 +459,37 @@ function _notifySwitchedAccountElsewhere() {
 		message,
 	});
 }
-
+function autoSwitcher(thing) {
+	const loggedInUsername = loggedInUser();
+	if (!loggedInUsername) return;
+	const currentUserIsAuthor = thing.getAuthor() === loggedInUsername;
+	if (!currentUserIsAuthor && !isAutoSwitched) {
+		sessionStorage.setItem('RESAccountSwitcherAutoSwitched', thing.isPost() ? 'post' : 'comment');
+		switchTo(thing.getAuthor());
+	}
+	if (isAutoSwitched === 'post' && currentUserIsAuthor && !autoSwitchNotificationShown) {
+		Notifications.showNotification({
+			moduleID: module.moduleID,
+			notificationID: 'auto-switched',
+			optionKey: 'autoSwitchOnPost',
+			header: i18n('accountSwitcherAutoSwitchNotificationTitle'),
+			message: i18n('accountSwitcherAutoSwitchPostDesc', loggedInUsername),
+		});
+		autoSwitchNotificationShown = true;
+		sessionStorage.removeItem('RESAccountSwitcherAutoSwitched');
+	}
+	if (isAutoSwitched === 'comment' && currentUserIsAuthor && !autoSwitchNotificationShown) {
+		Notifications.showNotification({
+			moduleID: module.moduleID,
+			notificationID: 'auto-switched',
+			optionKey: 'autoSwitchOnComment',
+			header: i18n('accountSwitcherAutoSwitchNotificationTitle'),
+			message: i18n('accountSwitcherAutoSwitchCommentDesc', loggedInUsername),
```

Comment:
Since the only thing which differs from the above `post` notification is `message`, they can be combined: `message: i18n(isAutoSwitched === 'comment' 
 ? 'accountSwitcherAutoSwitchCommentDesc' : 'accountSwitcherAutoSwitchPostDesc', loggedInUsername),`